test/integration/payment_method_test.exs

```diff
@@ -0,0 +1,147 @@
+defmodule Braintree.Integration.PaymentMethodTest do
+  use ExUnit.Case, async: true
+
+  alias Braintree.Customer
+  alias Braintree.PaymentMethod
+  alias Braintree.PaymentMethodNonce
+  alias Braintree.Testing.Nonces
+  alias Braintree.Testing.CreditCardNumbers
+  
+  @moduletag :integration
+
+  test "create/1 fails when customer_id not provided" do
+    {:error, error} = PaymentMethod.create(%{
+      payment_method_nonce: Nonces.transactable
+    })
+
+    assert error.message == "Customer ID is required."
+  end
+
+  test "create/1 fails when payment_method_nonce not provided" do
+    {:ok, customer} = Customer.create(%{
+      first_name: "Bill",
+      last_name: "Gates"
+    })
+
+    {:error, error} = PaymentMethod.create(%{
+      customer_id: customer.id
+    })
+
+    assert error.message == "Nonce is required."
+  end
+
+  test "create/1 can create a payment method from an existing customer and fake nonce" do
+    {:ok, customer} = Customer.create(%{
+      first_name: "Bill",
+      last_name: "Gates"
+    })
+
+    {:ok, payment_method} = PaymentMethod.create(%{
+        customer_id: customer.id,
+        payment_method_nonce: Nonces.transactable
+      })
+
+    assert payment_method.card_type == "Visa"
+    assert payment_method.bin =~ ~r/^\w+$/
+  end
+
+  test "create/1 can create a payment method from a vaulted credit card nonce" do
+    {:ok, customer} = Customer.create(
+      first_name: "Rick",
+      last_name: "Grimes",
+      credit_card: %{
+        number: master_card,
+        expiration_date: "01/2016",
+        cvv: "100"
+      }
+    )
+
+    card = customer.credit_cards |> List.first
```

Comment:
Better to use `[card] = customer.credit_cards` to pattern match here, we expect exactly one card.